app/controllers/v0/claim_letters_controller.rb

```diff
@@ -8,11 +8,21 @@ class ClaimLettersController < ApplicationController
     Sentry.set_tags(feature: 'claim-letters')
     service_tag 'claim-status'
 
+    VBMS_LIGHTHOUSE_MIGRATION_STATSD_KEY_PREFIX = 'vbms_lighthouse_claim_letters_provider_error'
+
     def index
       docs = service.get_letters
       log_metadata_to_datadog(docs)
 
       render json: docs
+    rescue => e
+      StatsD.increment("#{VBMS_LIGHTHOUSE_MIGRATION_STATSD_KEY_PREFIX}.index.#{@api_provider}")
```

Comment:
This error handling is duplicated between both the index and show methods. We could extract this into a private method to reduce duplication and increase maintainability.

It also uses the rails `action_name` making it more flexible.

replace with:
```
log_api_provider_error(e)
```
add private method:

```
def log_api_provider_error(error)
  metric_key = "#{VBMS_LIGHTHOUSE_MIGRATION_STATSD_KEY_PREFIX}.#{action_name}.#{@api_provider}"
  StatsD.increment(metric_key)
  ::Rails.logger.info("#{metric_key} error", {
    message_type: 'cst.api_provider.error',
    error_type: error.class.to_s,
    error_backtrace: error.backtrace&.first(3),
    api_provider: @api_provider
  })
end
```

Thoughts?

Reply:
sure! let's do it